mmv1/third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb

```diff
@@ -1004,6 +1063,131 @@ resource "google_cloudfunctions_function" "function" {
 `, projectNumber, networkName, vpcConnectorName, vpcConnectorName, vpcIp, bucketName, zipFilePath, functionName, vpcConnectorName)
 }
 
+<% unless version == "ga" -%>
+func testAccCloudFunctionsFunction_docker_repository(arRepoName, functionName, bucketName, zipFilePath string) string {
+	return fmt.Sprintf(`
+data "google_project" "project" {
+}
+
+resource "google_artifact_registry_repository" "test-ar-repo" {
+  repository_id = "%s"
+  location = "us-central1"
+  format = "DOCKER"
+}
+
+resource "google_artifact_registry_repository_iam_binding" "binding" {
+  location = google_artifact_registry_repository.test-ar-repo.location
+  repository = google_artifact_registry_repository.test-ar-repo.name
+  role = "roles/artifactregistry.admin"
+  members = [
+    "serviceAccount:service-${data.google_project.project.number}@gcf-admin-robot.iam.gserviceaccount.com",
+  ]
+}
+
+resource "google_storage_bucket" "bucket" {
+  name     = "%s"
+  location = "US"
+}
+
+resource "google_storage_bucket_object" "archive" {
+  name   = "index.zip"
+  bucket = google_storage_bucket.bucket.name
+  source = "%s"
+}
+
+resource "google_cloudfunctions_function" "function" {
+  name                  = "%s"
+  description           = "Function deployed to customer-provided Artifact Registry"
+  runtime               = "nodejs10"
+  available_memory_mb   = 128
+  source_archive_bucket = google_storage_bucket.bucket.name
+  source_archive_object = google_storage_bucket_object.archive.name
+  docker_repository     = google_artifact_registry_repository.test-ar-repo.id
+  trigger_http          = true
+  timeout               = 61
+  entry_point           = "helloGET"
+}
+`, arRepoName, bucketName, zipFilePath, functionName)
+}
+<% end -%>
+
+<% unless version == "ga" -%>
+func testAccCloudFunctionsFunction_cmek(kmsSuffix, arRepoName, functionName, bucketName, zipFilePath string) string {
+	return fmt.Sprintf(`
+data "google_project" "project" {
+}
+
+resource "google_kms_key_ring" "gcf_cmek_key_ring" {
```

Comment:
Because of issues with deleting kms keys, we need to "bootstrap" them instead of defining them in the terraform code. [Example](https://github.com/hashicorp/terraform-provider-google/blob/80f6dd2fcc1c209ed2b066d9b758db2e34145368/google/resource_pubsub_topic_test.go#L45)

Reply:
Yeah, KMS keyrings aren't deletable, so I thought that's handled by project rental. Re-factored this to use the suggested util instead.